mlir/utils/jenkins/Jenkinsfile

```diff
@@ -793,5 +817,79 @@ pipeline {
                 }
             }
         }
+        stage ("MIGraphX") {
+            when {
+                beforeAgent true;
+                equals expected: true, actual: params.checkMIGraphX;
+                equals expected: true, actual: params.nightly;
+            }
+            matrix {
+                axes {
+                    axis {
+                        name 'CODEPATH'
+                        values 'mfma', 'vanilla', 'navi21'
+                    }
+                }
+                agent {
+                    docker {
+                        image dockerImage()
+                        args dockerArgs()
+                        label getLabelFromCodepath("${CODEPATH}")
+                        alwaysPull true
+                    }
+                }
+                when {
+                    beforeAgent true
+                    expression { return shouldRunFromCodepath("$CODEPATH") }
+                }
+                environment {
+                    HOME="${WORKSPACE}"
+                }
+                stages {
+                    stage('Environment') {
+                        steps {
+                            echo "codepath is ${CODEPATH}"
+                            showEnv()
+                        }
+                    }
+                    stage("Install MIGraphX Dependencies") {
+                        steps {
+                            // We are installing miopen-hip here (and not in Dockerfile) to avoid collision with MIOpen checks above
+                            sh 'apt-get install miopen-hip'
+                            // Package and install current checkout of rocMLIR as MIGraphX dependency.
+                            sh 'cget -p ${WORKSPACE}/MIGraphXDeps install ${WORKSPACE} -DBUILD_MIXR_TARGET=On -DLLVM_ENABLE_ZSTD=Off -DLLVM_ENABLE_THREADS=Off'
+                        }
+                    }
+                    stage("Build MIGraphX with MLIR") {
+                        steps {
+                            sh 'rm -rf MIGraphX'
+                            dir('MIGraphX') {
+                                getAndBuildMIGraphX("-DCMAKE_PREFIX_PATH='/MIGraphXDeps;${WORKSPACE}/MIGraphXDeps' -DMIGRAPHX_ENABLE_MLIR=On -DCMAKE_CXX_COMPILER=/opt/rocm/llvm/bin/clang++")
+                            }
+                        }
+                    }
+                    stage("Verify MIGraphX with MLIR") {
+                        steps {
+                            dir('MIGraphX/build') {
+                                // Verify MLIR unit tests
+                                sh 'make -j$(nproc) test_gpu_mlir'
+                                sh 'ctest -R test_gpu_mlir'
+                                // Verify ResNet50
```

Comment:
If we remove this, can we not install MIOpen?

Reply:
No, MIOpen is first class dependency which we cant disable atm.